### PR TITLE
Image url submission

### LINKED ISSUE #2: Save Image Link
#### As a user I want to save a link to an image.

__Story__:

Now that we have an awesome landing page for our users to experience it is time
to introduce our first feature! In general, we want to deliver value to our
customers as fast as possible. As a result, we have decided that we want our
users to be able to save links to images, and immediately see that we have
saved the URL to the image.

One could argue that simply saving the image, and not showing it, provides some
value. Yes, you could split this story here. However, because saving image URLs
and viewing those images are both relatively small additions that work well
together, it is acceptable to complete them together. Note, on the other hand,
that we do not yet want to implement the image index. Its addition adds too
much complexity, so we’ll save that for a future story.

In general, feature addition, or stories, should be split into the smallest
deliverable chunk that provides value to your users with the caveat that it
might make sense to combine two, sometimes more, tiny and related features into
a single story.

__Acceptance criteria__:
- [x] The landing page links to the image link submission form.
- [x] The link is entered through a form.
- [x] After the form is saved, the link is persisted in the database.
- [x] After the form is submitted, the user is redirected to a page displaying
  the image.

__Dependencies__:
- Deployment Environment

__Note__: Many image sharing sites will permit one to upload images, or
download images when provided a link. For this project we will only work with
the link, i.e., URL, to various images.

### PR DESCRIPTION
**Pull request by dannycho7**
Closes #2

### CODE REVIEW

File: app/controllers/images_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  def new
+    # renders the form to submit the file upload
+    @image = Image.new
+  end
+
+  def create
+    # saves the image in the database
+    @image = Image.new
+    @image.url = params[:image]['url']
```
Comment (mlewisno) [inline app/controllers/images_controller.rb]: Is there a way you can collapse those two lines into one? What would happen if there is no `:image` key in the params hash? How would the controller handle that?

File: db/migrate/20170627215251_create_images_table_with_columns.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class CreateImagesTableWithColumns < ActiveRecord::Migration[5.0]
+  def change
```
Comment (mlewisno) [inline db/migrate/20170627215251_create_images_table_with_columns.rb]: In general we try to make our commits atomic, meaning that the migration that adds a table should be in the same commit as the ruby file that defines its class.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -3,7 +3,7 @@
   <head>
     <title>Base</title>
     <%= csrf_meta_tags %>
-
+    <link href="https://fonts.googleapis.com/css?family=Josefin+Sans" rel="stylesheet">
```
Comment (mlewisno) [inline app/views/layouts/application.html.erb]: Nice!

File: app/views/welcome/index.html.erb [hunk 4]
```diff
@@ -1,2 +1,12 @@
 <h1>Welcome to the image sharing ROPES project</h1>
-<%= link_to "Add a new image url", new_image_path %>
\ No newline at end of file
+<%= link_to "Add a new image url", new_image_path %>
+
+<h1>Here are some sample images that were uploaded most recently</h1>
+
+<div>
+	<% if @images %>
```
Comment (mlewisno) [inline app/views/welcome/index.html.erb]: Is there ever a case where the index should be shown if `@images` isn't defined?

Comment (dannycho7) [inline app/views/welcome/index.html.erb]: I changed the code to this. 


<div>
	<% if @images %>
		<h1>Here are some sample images that were uploaded most recently</h1>
		<% @images.each do |image| %>
			" />
		<% end %>
	<% else %>
		<h1>There are currently no images uploaded</h1>
	<% end %>
</div>

File: app/models/image.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+class Image < ApplicationRecord
+  validates :url, presence: true # Make sure the owner's name is present.
```
Comment (dannycho7) [inline app/models/image.rb]: The comment was wrong. I added it at the very beginning and completely forgot about it.

File: app/controllers/images_controller.rb (hunk 1 above)
Comment (dannycho7) [inline app/controllers/images_controller.rb]: I'm not exactly sure how to accomplish everything in one line. Is there a ruby keyword shortcut for creating models?

Comment (mlewisno) [inline app/controllers/images_controller.rb]: Yeah! You can do `Image.new(url: params[:image]['url'])`!

**Pull request merged**